facebook/OpenBIC
# fby35: nf: Initialization GPIO

Summary:
- Initialization GPIO.
- Disable GPIO internal pull down of BIC.

Note:
- Reference GPIO table 20230322_1800.

Test plan:
- Build cdoe: Pass
- Check GPIO table setting: Pass

Log:
1. Check GPIO table. uart:~$ platform gpio list_all
[0  ] PWR_ON_RST_N_R                     : OD  | output(I) | 0(0)
[1  ] SYS_RST_N_R                        : OD  | output(I) | 0(0)
[2  ] PERST_ASIC_N_R                     : OD  | output(I) | 0(0)
[3  ] JTAG2_NTRST2_R                     : PP  | output(O) | 0(0)
[4  ] RST_PCIE_MB_EXP_N                  : PP  | input (I) | 0(0)
[5  ] AUX_PWR_EN_4C                      : PP  | input (I) | 0(0)
[6  ] PG_CARD_OK                         : PP  | output(O) | 0(0)
[7  ] POWER_EN_R                         : PP  | input (I) | 0(0)
[9  ] MASTER                             : PP  | output(O) | 0(0)
[10 ] MUX_OE                             : PP  | output(O) | 0(0)
[11 ] SMB_12V_INA_ALRT_N                 : PP  | input (I) | 1(1)
[12 ] SMB_3V3_INA_ALRT_N                 : PP  | input (I) | 0(0)
[13 ] SPI_RST_FLASH_N_R                  : OD  | output(I) | 1(1)
[14 ] SMBUS_ALERT_R_N                    : OD  | output(I) | 1(1)
[15 ] JTAG2_ASIC_PORT_SEL_R_EN           : PP  | output(O) | 0(0)
[16 ] P0V85_BIC_ASIC_EN                  : PP  | output(O) | 0(0)
[17 ] PWRGD_P0V85_ASIC                   : PP  | input (I) | 0(0)
[18 ] P1V2_BIC_ASIC_EN                   : PP  | output(O) | 0(0)
[19 ] PWRGD_P1V2_ASIC                    : PP  | input (I) | 0(0)
[20 ] P1V8_BIC_ASIC_EN                   : PP  | output(O) | 0(0)
[21 ] PWRGD_P1V8_ASIC                    : PP  | input (I) | 0(0)
[22 ] P0V8_BIC_ASIC_EN                   : PP  | output(O) | 0(0)
[23 ] PWRGD_P0V8_ASIC                    : PP  | input (I) | 0(0)
[24 ] PVPP_AB_BIC_DIMM_EN                : PP  | output(O) | 0(0)
[25 ] PWRGD_PVPP_AB                      : PP  | input (I) | 0(0)
[26 ] PVPP_CD_BIC_DIMM_EN                : PP  | output(O) | 0(0)
[27 ] PWRGD_PVPP_CD                      : PP  | input (I) | 0(0)
[28 ] PVTT_AB_BIC_DIMM_EN                : PP  | output(O) | 0(0)
[29 ] PWRGD_PVTT_AB                      : PP  | input (I) | 0(0)
[30 ] PVTT_CD_BIC_DIMM_EN                : PP  | output(O) | 0(0)
[31 ] PWRGD_PVTT_CD                      : PP  | input (I) | 0(0)
[32 ] JTAG2_BIC_SHIFT_EN_R               : PP  | output(O) | 0(0)
[33 ] SPI_BIC_SHIFT_EN_R                 : OD  | output(I) | 0(0)
[34 ] USB_MUX2_SEL                       : PP  | output(O) | 0(0)
[35 ] PAD_CORE_TAP_CTRL_L_R              : PP  | output(O) | 1(1)
[37 ] SMB_HOST_MUX_PMIC_IN_R             : PP  | output(O) | 1(1)
[38 ] SMB_HOST_MUX_DIMM_IN_R             : PP  | output(O) | 0(0)
[40 ] SLOT_ID0                           : PP  | input (I) | 0(0)
[41 ] SLOT_ID1                           : PP  | input (I) | 1(1)
[42 ] BOARD_ID0                          : PP  | input (I) | 0(0)
[43 ] BOARD_ID1                          : PP  | input (I) | 1(1)
[44 ] BOARD_ID2                          : PP  | input (I) | 0(0)
[45 ] BOARD_ID3                          : PP  | input (I) | 0(0)
[46 ] USB_MUX1_R_OE                      : PP  | output(O) | 1(1)
[47 ] USB_MUX2_R_OE                      : PP  | output(O) | 1(1)
[50 ] PWRGD_P1V2_STBY_R                  : PP  | input (I) | 1(1)
[51 ] PVDDQ_AB_EN_BIC                    : PP  | output(O) | 0(0)
[52 ] PWRGD_PVDDQ_AB_BIC                 : PP  | input (I) | 1(1)
[53 ] PVDDQ_CD_EN_BIC                    : PP  | output(O) | 0(0)
[54 ] PWRGD_PVDDQ_CD_BIC                 : PP  | input (I) | 1(1)
[55 ] P0V75_ASIC_EN_BIC                  : PP  | output(O) | 0(0)
[56 ] PWRGD_P0V75_ASIC_BIC               : PP  | input (I) | 0(0)
[57 ] P5V_STBY_EN_BIC                    : PP  | output(O) | 0(0)
[58 ] PWRGD_P5V_STBY_BIC                 : PP  | input (I) | 0(0)
[59 ] SMB_BIC_PMIC_ALERT_N_R             : PP  | input (I) | 0(0)
[68 ] LS_SMB_BIC_ASIC_EN                 : PP  | output(O) | 0(0)
[69 ] LS_UART_BIC_ASIC_EN                : PP  | output(O) | 0(0)
[70 ] LS_BIC_ASIC_EN                     : PP  | output(O) | 0(0)
[71 ] SMB_SENSOR_ALERT_N_R               : PP  | input (I) | 1(1)
[78 ] ADAPTER_SEL_R                      : PP  | input (I) | 1(1)
[90 ] LED_CXL_POWER                      : PP  | output(O) | 0(0)
[91 ] FM_BOARD_REV_ID2                   : PP  | input (I) | 0(0)
[92 ] FM_BOARD_REV_ID1                   : PP  | input (I) | 0(0)
[93 ] FM_BOARD_REV_ID0                   : PP  | input (I) | 0(0)
[94 ] IOE_RST_N                          : OD  | output(I) | 0(0)
[95 ] IOE_INT_N_R                        : PP  | input (I) | 1(1)
[96 ] BIC_SECUREBOOT                     : PP  | input (I) | 0(0)
[97 ] LED_CXL_FAULT                      : OD  | output(O) | 0(0)
[98 ] FM_PWRBRK_PRIMARY_R_N              : PP  | input (I) | 1(1)
[99 ] BIC_ESPI_SELECT                    : PP  | input (I) | 0(0)
[100] UART_BIC_R_RXD                     : PP  | output(O) | 0(0)
[101] UART_BIC_R_TXD                     : PP  | input (I) | 1(1)
[107] CLK_100M_OSC_EN                    : PP  | output(O) | 0(0)
[129] LED_ASIC_R_HB                      : PP  | input (I) | 1(1)
[130] LED_ASIC_R_1                       : PP  | input (I) | 1(1)
[131] LED_ASIC_R_2                       : PP  | input (I) | 1(1)
[132] LED_ASIC_R_3                       : PP  | input (I) | 1(1)
